### PR TITLE
fix: input counter issue on Input

### PR DESCRIPTION
**Pull request by MrSltun**
### Description
This PR updates the input character counter to fix an issue that happens when we update the Input's value using React's state

### Videos

| Before | After |
|---|---|
| <video src="https://github.com/artsy/palette/assets/20655703/fd845028-0352-469e-92d9-f219633b7320" /> | <video src="https://github.com/artsy/palette/assets/20655703/3c142cd2-4dd0-443b-ad91-5981971c6b91" /> |

### CODE REVIEW

File: packages/palette/src/elements/Input/Input.tsx [hunk 1]
```diff
@@ -45,16 +45,23 @@ export const Input: React.ForwardRefExoticComponent<
...
     },
     ref
   ) => {
     const [boxProps, inputProps] = splitBoxProps(rest)
-    const [charCount, setCharCount] = React.useState(0)
-
-    const countChars = (e: React.ChangeEvent<HTMLInputElement>) => {
-      setCharCount(e.target.value.length)
-    }
+    const [value, setValue] = React.useState(inputProps.value || defaultValue)
```
Comment (dzucconi) [inline packages/palette/src/elements/Input/Input.tsx:55]: Bifurcating the state like this seems excessive and is going to wind up with more bugs down the line. There's a simpler fix IMO

Comment (dzucconi) [reply packages/palette/src/elements/Input/Input.tsx:55]: I guess this is what `Textarea` does, so at least there's precedent. I suspect though there's a way to externalize the notion of character limit and use it in both simultaneously.

**Review by dzucconi — APPROVED** [review]

File: packages/palette/src/elements/Input/Input.tsx (hunk 1 above)
Comment (MrSltun) [reply packages/palette/src/elements/Input/Input.tsx:55]: Would love to hear about the simpler fix :)

**Comment by artsyit** [comment]
:rocket: PR was released in [`@artsy/palette-charts@37.6.0`](https://github.com/artsy/palette/releases/tag/%40artsy/palette-charts%4037.6.0), [`@artsy/palette@38.6.0`](https://github.com/artsy/palette/releases/tag/%40artsy/palette%4038.6.0) :rocket:

**Pull request merged**